File: Unit 1 Fundamentals of programming/Fundamentals Topic 4 Arrays/Topic 4 VB Programs/W4 Q6 Car Park.vb
Module Module1




    Sub Main()

        'W4 Q6 Car Park
        Dim carPark(9, 5) As String
        Dim carReg As String = ""
        Dim row, column As Integer
        Dim emptySpace, rowValid, columnValid As Boolean
        emptySpace = False

        For row = 0 To 9
            For column = 0 To 5
                carPark(row, column) = "empty"
            Next
        Next
        carPark(0, 0) = "taken"

        'display car park grid
        For row = 0 To 9
            For column = 0 To 5
                Console.Write(carPark(row, column) & vbTab)
            Next
            Console.WriteLine()
        Next

        '--------------------------------------------------------------------------------
        'Now we will attempt to assign a parking space

        carReg = InputBox("Enter registration")

        While emptySpace = False


            row = InputBox("Enter row")
            While row < 1 Or row > 10 'Ensure row is a valid range
                row = InputBox("Error enter a number from 1 to 10")
            End While
            row = row - 1 'subtract 1 because the array index starts at 0 and not 1
            rowValid = True 'set the row valid flag to true

            column = InputBox("Enter a column 1 to 6") 'get a valid column number
            While column < 1 Or column > 6
                column = InputBox("Error enter a column from 1 to 6")
            End While
            column = column - 1
            columnValid = True 'set the column valid flag to true
            If rowValid And columnValid And carPark(row, column) = "empty" Then
                emptySpace = True 'set the empty space to true, this space is available
            Else
                'reset the flags to false, re-intialize row and column to 100 and output message
                rowValid = False
                columnValid = False
                row = 100
                column = 100
                MsgBox("That space is taken")
            End If

        End While
        carPark(row, column) = carReg 'if coordinates are valid and space is available then assign space

        'display car park grid
        Console.Clear() 'clearing the last screen
        For row = 0 To 9
            For column = 0 To 5
                Console.Write(carPark(row, column) & vbTab)
            Next
            Console.WriteLine()
        Next

        MsgBox("End")
    End Sub





End Module

File: Unit 1 Fundamentals of programming/Fundamentals Topic 4 Arrays/Topic 4 Python Programs/W4 Q6 car park.py
#AS AQA unit 1 W4 Q6 car park
carPark = []
for row in range(0,10):
    carPark.append([])
    for column in range(0,6):
        carPark[row].append("empty    ")
#next statement is for test purposes
carPark[0][0] = "TAKEN    "
        
for row in range(0,10):
    for column in range(0,6):
        symbol = carPark[row][column]
        print (symbol,' ', end='')
    print()


print("Enter the car registration: ")
regNo = input()
#now specify a position where car is parked
row = int(input("Enter row where car is parked (1-10): "))
column = int(input ("Enter column where car is parked (1-6): "))
emptySpace = False

while emptySpace == False:

    while row < 1 or row > 10:
        row = int(input("Row must be between 1 and 10 - please re-enter: ")) 
    row = row - 1
    rowValid = True
    
    while column < 1 or column > 6:
        column = int(input("Column must be between 1 and 6 - please re-enter: "))
    column = column - 1
    columnValid = True
    if rowValid and columnValid and carPark[row][column]== "empty    ":
        emptySpace = True
    else:
        rowValid = False
        columnValid = False
        print ("\nThat space is taken\n")  #\n goes to next line
carPark[row][column] = regNo

for row in range(0,10):
    for column in range(0,6):
        symbol = carPark[row][column]
        print (symbol,' ', end = '')       #end ='' stays on same line
    print()

input("\nPress Enter to end ")